src/auth.rs

```diff
@@ -30,78 +33,98 @@ pub enum Action {
 
+        // get token from headers
+        let token = match TezosAuthorizationString::extract(auth_data) {
+            Ok(t) => t,
+            Err(e) => return Outcome::Failure((Status::Unauthorized, e)),
+        };
         match token.action() {
             // content actions have the same authz process
             Action::Put { orbit_id, .. }
             | Action::Get { orbit_id, .. }
             | Action::List { orbit_id }
             | Action::Del { orbit_id, .. } => {
-                let read_orbits = orbits.orbits().await;
-                let orbit = match read_orbits.get(orbit_id) {
+                let orbit = match load_orbit(config.database.path, orbit_id) {
                     Some(o) => o,
                     None => {
-                        return Outcome::Failure((
-                            Status::Unauthorized,
-                            anyhow!("No authorization policy found"),
-                        ))
+                        return Outcome::Failure((Status::Unauthorized, anyhow!("No Orbit found")))
                     }
                 };
                 match orbit.auth().authorize(&token).await {
-                    Ok(_) => Outcome::Success(AuthWrapper(token)),
+                    Ok(_) => Outcome::Success(orbit),
                     Err(e) => Outcome::Failure((Status::Unauthorized, e)),
                 }
             }
             // Create actions dont have an existing orbit to authorize against, it's a node policy
             // TODO have policy config, for now just be very permissive :shrug:
-            Action::Create { .. } => match req.rocket().state::<T::Policy>() {
-                Some(auth) => match auth.authorize(&token).await {
-                    Ok(_) => Outcome::Success(AuthWrapper(token)),
-                    Err(e) => Outcome::Failure((Status::Unauthorized, e)),
-                },
-                None => Outcome::Success(AuthWrapper(token)),
+            Action::Create {
+                orbit_id,
+                parameters,
+                ..
+            } => match TezosBasicAuthorization.authorize(&token).await {
+                Ok(_) => {
+                    if let Err(e) = verify_oid(orbit_id, &token.pkh, parameters) {
+                        Outcome::Failure((Status::BadRequest, "Incorrect Orbit ID"))
+                    }
+                    let vm = DIDURL {
+                        did: format!("did:pkh:tz:{}", &token.pkh),
+                        fragment: Some("TezosMethod2021".to_string()),
+                        ..Default::default()
+                    };
+                    match create_orbit(
+                        orbit_id,
+                        config.database.path,
+                        TezosBasicAuthorization,
+                        vec![vm],
+                        auth_data.as_bytes(),
+                    )
+                    .await
+                    {
+                        Ok(orbit) => Outcome::Success(orbit),
```

Comment:
I realized, this will allow a create message to succeed when sent to any route. because of auth, this is not really "dangerous", as there's materially no difference between the routes, but it is probably not intended.